src/main/java/com/iridium/iridiumskyblock/gui/IslandCustomAmountBankGUI.java

```diff
@@ -0,0 +1,103 @@
+    private final Map<UUID, Data> opened = new HashMap();
+
+    public IslandCustomAmountBankGUI() {
+        if (VirtualAnvil.getVirtualAnvil() == null)
+            VirtualAnvil.load();
+
+        Bukkit.getPluginManager().registerEvents(this, IridiumSkyblock.getInstance());
+    }
+
+    public void openInventory(Player player, Island island, BankItem bankItem, boolean deposit) {
+        VirtualAnvil.getVirtualAnvil().openAnvil(player);
+        IslandBank islandBank = IridiumSkyblock.getInstance().getIslandManager().getIslandBank(island, bankItem);
+        player.getOpenInventory().getTopInventory().setItem(0, ItemStackUtils.makeItem(bankItem.getItem(), Collections.singletonList(new Placeholder("amount", IridiumSkyblock.getInstance().getNumberFormatter().format(islandBank.getNumber())))));
+        this.opened.put(player.getUniqueId(), new Data(island, deposit, bankItem));
+    }
+
+    @EventHandler
+    public void onInvClose(InventoryCloseEvent e) {
+        if (this.opened.containsKey(e.getPlayer().getUniqueId())) {
+            e.getInventory().setItem(0, null);
+            this.opened.remove(e.getPlayer().getUniqueId());
+        }
+    }
+
+    @EventHandler
+    public void onClick(InventoryClickEvent e) {
+        if (!this.opened.containsKey(e.getWhoClicked().getUniqueId()))
+            return;
+
+        e.setCancelled(true);
+
+        if (!(e.getClickedInventory() instanceof AnvilInventory))
+            return;
+
+        AnvilInventory anvilInventory = (AnvilInventory) e.getClickedInventory();
+
+        if (e.getSlot() == 2) {
+            if (e.getWhoClicked().getExpToLevel() >= anvilInventory.getRepairCost())
+                ((Player) e.getWhoClicked()).giveExpLevels(0);
+            Data data = this.opened.get(e.getWhoClicked().getUniqueId());
+            e.setCancelled(true);
+            if (data.isDeposit())
+                try {
+                    data.bankItem.deposit((Player) e.getWhoClicked(), Double.parseDouble(Objects.requireNonNull(anvilInventory.getRenameText())));
+                } catch (NullPointerException | NumberFormatException ignored) {
+                    e.getWhoClicked().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().notANumber.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                }
+            else try {
+                data.bankItem.withdraw((Player) e.getWhoClicked(), Double.parseDouble(Objects.requireNonNull(anvilInventory.getRenameText())));
+            } catch (NullPointerException | NumberFormatException ignored) {
+                e.getWhoClicked().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().notANumber.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+            }
+
+            e.getWhoClicked().openInventory(e.getInventory());
+            e.getWhoClicked().closeInventory();
+        }
+    }
+
+    private static final class Data {
```

Comment:
Should be using Lombok's @Getter and @AllArgsConstructor here

Reply:
Haven't worked much with Lombok, yet.

Will read into it though